client/src/message_popup/renderers/response_summary_coding.jsx

```diff
@@ -0,0 +1,223 @@
+
+import ReadMore from './read_more.jsx';
+
+/*
+Component that displays the summary of responses, either audio or text and asks user to label responses from multi-select tagging options
+*/
+export default class extends React.Component {
+
+  props: {
+    responses: Array<Object>,
+    onLogMessage: Function,
+    questionEl: Function,
+  };
+
+  static displayName = 'ResponseSummaryCoding';
+
+  static propTypes = {
+    responses: PropTypes.array.isRequired,
+    onLogMessage: PropTypes.func.isRequired,
+    children: PropTypes.node,
+  };
+
+
+
+  state = {
+    studentSourcedLabels: [],
+  }
+
+
+
+  // Supports audio and two forms of text response.
+  computeSummaryItems = () => {
+    const {responses} = this.props;
+
+    return _.compact(responses.map((response) => {
+      const questionId = response.question.id;
+      const questionText = response.question.text;
+
+      if (response.audioResponse) {
+        const audioUrl = response.audioResponse.downloadUrl;
+        return {questionId, questionText, audioUrl};
+      }
+
+      if (response.textResponse) {
+        const {responseText} = response.textResponse;
+        return {questionId, questionText, responseText};
+      }
+
+      if (response.responseText) {
+        const {responseText} = response;
+        return {questionId, questionText, responseText};
+      }
+
+      return null;
+    }));
+  };
+
+  getIndexOfQuestionId(questionId, labelCategory) {
+
+    var arr = this.state.studentSourcedLabels;
+    for (var i = 0; i < arr.length; i++) {
+      if (arr[i].questionId === questionId && arr[i].category === labelCategory) {
+        return i;
+      }    
+    }
+    return -1;
+  }
+  getValueByQuestionId (questionId, labelCategory) {
+    var retVal = "";
+    var arr = this.state.studentSourcedLabels; 
+    var labelIndex = this.getIndexOfQuestionId(questionId, labelCategory);
+
+    if (labelIndex > -1) {
+      retVal = arr[labelIndex].label ;
+    } 
+
+    return retVal;
+  }
+
+
+
+  handleSelectionChange(questionId, labelCategory, event) {   
+    var Label = {
+      questionId:questionId, 
+      category:labelCategory, 
+      label:event.target.textContent, 
+    };
+
+    var newStudentSourcedLabels = _.cloneDeep(this.state.studentSourcedLabels);
+    var labelIndex = this.getIndexOfQuestionId(questionId, labelCategory);
+
+    if (labelIndex > -1) {
+      //update      
+      newStudentSourcedLabels[labelIndex] = Label;
+    } else {
+      //add
+      newStudentSourcedLabels = this.state.studentSourcedLabels.concat([Label]);
+    }
+
+    this.setState({studentSourcedLabels: newStudentSourcedLabels});
+    this.onLogMessageWithQuestionCoding(questionId, 'student_coding', newStudentSourcedLabels);
+  }
+  // Mixes in question to payload
+  onLogMessageWithQuestionCoding = (questionId, type, coding) => {
+    this.props.onLogMessage(type, {
+      questionId, 
+      coding
+    });
+  };
+
+
+  render() {
```

Comment:
hi @TheGarron!  i commented on your pull request and you should get a notification of it :)